### PR TITLE
Added Flowers

### PR DESCRIPTION
**Pull request by BoredSathvik**
Pls check my spellings ;( i am bad at it

### CODE REVIEW

**Comment by Bavfalcon9** [comment]
Is a new class for every type of flower really necessary 🤔

**Comment by Bavfalcon9** [comment]
Maybe something like
```ts
class Flower {
  public static RedTulip: Flower = new Flower('minecraft:redtulip');
 ```
 would suffice

**Comment by BoredSathvik** [comment]
I mean all the blocks are in separate classes so i did the same :)

**Comment by Jakobg1215** [comment]
> Is a new class for every type of flower really necessary 🤔

Yes, each flower is different and need there own class.

**Comment by BoredSathvik** [comment]
I would prefer doing the making a new class too but all the blocks are in different class so i followed the same thing uwu

**Comment by Bavfalcon9** [comment]
> 
> 
> > Is a new class for every type of flower really necessary 🤔
> 
> Yes, each flower is different and need there own class.

It's not, this code is repetitive, and can be added as a property rather than this implementation.
Something like 
```ts
class Flower {
 public static new(type: FlowerType): Flower {}
 }
 ``` should, and would suffice.

**Comment by Jakobg1215** [comment]
> > > Is a new class for every type of flower really necessary 🤔
> > 
> > 
> > Yes, each flower is different and need there own class.
> 
> It's not, this code is repetitive, and can be added as a property rather than this implementation.
> Something like
> 
> ```ts
> class Flower {
>  public static new(type: FlowerType): Flower {}
>  }
>  ``` should, and would suffice.
> ```

But why?

**Comment by Bavfalcon9** [comment]
It's a flower variety, it's the same way with stone, stone could have implementation where you could get the type of stone you want. eg: `Stone.Andesite` (or related).

**Comment by filfat** [comment]
@Bavfalcon9 preperation for the big flattening where each meta value *is* a different block

**Comment by Bavfalcon9** [comment]
> 
> 
> @Bavfalcon9 preperation for the big flattening where each meta value _is_ a different block

Still would be clean implementation 😄

**Comment by filfat** [comment]
> > @Bavfalcon9 preperation for the big flattening where each meta value _is_ a different block
> 
> Still would be clean implementation 😄

Again, after the big flattening they are completely different blocks with completely different IDs (see Java edition). This is just being future proof.

**Comment by BoredSathvik** [comment]
so is this pull getting accepted, or I need to add more stuff to this pull 😞

**Review by Bavfalcon9 — APPROVED** [review]
Looks good, 😄

File: src/block/blocks/Poppy.ts [hunk 1]
```diff
@@ -3,16 +3,14 @@ import Flowable from '../Flowable';
...
-    CornFlower,
-    LilyOfTheValley
+    BlueOrchid = 1,
+    Allium = 2,
+    AzureBluet = 3,
+    RedTulip = 4,
+    OrangeTulip = 5,
+    WhiteTulip = 6,
+    PinkTulip = 7,
+    OxeyeDaisy = 8
```
Comment (Bavfalcon9) [inline src/block/blocks/Poppy.ts:13]: Typescript automatically increments enums, please revert this change 😄

**Review by Bavfalcon9 — CHANGES_REQUESTED** [review]

File: src/block/blocks/Poppy.ts (hunk 1 above)
Comment (Bavfalcon9) [reply src/block/blocks/Poppy.ts:13]: (forgot this file sorry 😂)

**Review by BoredSathvik — COMMENTED** [review]
Done

**Review by Bavfalcon9 — APPROVED** [review]

File: src/block/blocks/Poppy.ts (hunk 1 above)
Comment (filfat) [reply src/block/blocks/Poppy.ts]: @Bavfalcon9 Yes it does but we've been explicit about enums everywhere else so would be weird to only have it auto increments here.

Comment (Bavfalcon9) [reply src/block/blocks/Poppy.ts]: requested mainly caused enums were removed
